config/config.go

```diff
@@ -848,6 +852,211 @@ func (c *Config) LoadConfigData(data []byte) error {
+				if err != nil {
+					return fmt.Errorf("Couldn't serialize config")
+				}
+
+			} else {
+				// Just serialize
+				kv := val.(*ast.KeyValue)
+
+				_, err := f.WriteString(tabChars + "  " + kv.Key + "=" + kv.Value.Source() + "\n")
+				if err != nil {
+					return fmt.Errorf("Couldn't serialize config")
+				}
+			}
+		}
+	}
+
+	// Process subtables next
+	for key := range t.Fields {
+		val := t.Fields[key]
+		switch val.(type) {
+		case *ast.Table:
+			// Recurse to handle indefinitely nested AST
+			tbl := val.(*ast.Table)
+			err := serializeTable(tbl, config, f, parentTableName+t.Name+".", numTabs+1, false)
+			if err != nil {
+				return fmt.Errorf("Couldn't serialize config")
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *Config) serializePlugin(pluginName string, pluginType string, f *os.File) error {
+	// Serialize table header
+	tableHeader := "[[" + pluginType + "." + pluginName + "]]\n"
+	creator := inputs.Inputs[pluginName]
+	input := creator()
+	config := input.SampleConfig()
+
+	_, err := f.WriteString(tableHeader)
+	if err != nil {
+		return fmt.Errorf("Couldn't serialize config")
+	}
+	_, err = f.WriteString(config)
+	if err != nil {
+		return fmt.Errorf("Couldn't serialize config")
+	}
+	return nil
+}
+
+func (c *Config) serializeConfig(data []byte, newConfigPath string, config map[string]interface{}, pluginName string, pluginType string, operationType string) error {
+	f, _ := os.OpenFile(newConfigPath, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+	f.Truncate(0)
+	defer f.Close()
+
+	tbl, err := parseConfig(data)
+	if err != nil {
+		return fmt.Errorf("Error parsing data: %s", err)
+	}
+
+	configParam := map[string]interface{}{}
+
+	// Parse tags tables first:
+	for _, tableName := range []string{"tags", "global_tags"} {
+		if val, ok := tbl.Fields[tableName]; ok {
+			subTable, ok := val.(*ast.Table)
+			if !ok {
+				return fmt.Errorf("invalid configuration, bad table name %q", tableName)
+			}
+
+			err := serializeTable(subTable, configParam, f, "", 0, false)
+			if err != nil {
+				return fmt.Errorf("Couldn't serialize config")
+			}
+
+			_, err = f.WriteString("\n")
+			if err != nil {
+				return fmt.Errorf("Couldn't serialize config")
+			}
+		}
+	}
+
+	// Parse agent table:
+	if val, ok := tbl.Fields["agent"]; ok {
+		subTable, ok := val.(*ast.Table)
+		if !ok {
+			return fmt.Errorf("invalid configuration, error parsing agent table")
+		}
+
+		err := serializeTable(subTable, configParam, f, "", 0, false)
+		if err != nil {
+			return fmt.Errorf("Couldn't serialize config")
+		}
+
+		_, err = f.WriteString("\n")
+		if err != nil {
+			return fmt.Errorf("Couldn't serialize config")
+		}
+	}
+
+	// Parse all the rest of the plugins:
+	for pType, val := range tbl.Fields {
+		subTable, ok := val.(*ast.Table)
+		if !ok {
+			return fmt.Errorf("invalid configuration, error parsing field %q as table", pType)
+		}
+
+		switch pType {
+		case "agent", "global_tags", "tags":
```

Comment:
Why is there an empty case here?

Reply:
@nikhilJain17 can maybe confirm but I believe it's because we don't want to serialize those sub tables since we handled them earlier

Reply:
You could replace this case with a default case or just remove it, and do one if-statement for the tables that you do want to serialize

Reply:
Yeah as Jodie said, you can just remove them all together. For a switch statement, if a case is not met, it just won't go into any of the code blocks.